### PR TITLE
GIX-1336: Refactor sns participation flow 1

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Simplify the payment flow code to make it more maintainable.

# Changes

Main change
* Remove access to ticketStore in `participateInSnsSwap` and pass ticket as parameter.

Minor change
* Started some renaming from "Sale" to "Swap" for consistency.

# Tests

* Fixed tests after the refactor and rename

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/lib/services/sns-sale.services.spec.ts [hunk 1]
```diff
@@ -1259,22 +1213,24 @@ describe("sns-api", () => {
...
             "The token amount can only be refreshed when the canister is in the OPEN state"
           )
         );
         spyOnSendICP.mockRejectedValue(new TxTooOldError(0));
 
-        await participateInSnsSale({
+        snsTicketsStore.setTicket({
+          rootCanisterId: rootCanisterIdMock,
+          ticket: testTicket,
+        });
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/sns-sale.services.spec.ts:1226]: Why do you still need this?

And if you still need it, can you keep it in the same place where it was?

Comment (lmuntaner) [reply frontend/src/tests/lib/services/sns-sale.services.spec.ts:1226]: Yes, this test checks that there is no ticket in the store after the process.

I'll revert to the same code.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]

**Pull request merged**